src/test/ui/const-generics/const-param-type-depends-on-type-param-ungated.stderr

```diff
@@ -7,7 +7,7 @@ LL | struct B<T, const N: T>(PhantomData<[T; N]>);
    = note: see issue #44580 <https://github.com/rust-lang/rust/issues/44580> for more information
    = help: add `#![feature(const_generics)]` to the crate attributes to enable
 
-error[E0741]: the types of const generic parameters must derive `PartialEq` and `Eq`
+error[E0741]: `T` must be annotated with `#[derive(PartialEq, Eq)]` to be used as the type of a const parameter
   --> $DIR/const-param-type-depends-on-type-param-ungated.rs:3:22
    |
 LL | struct B<T, const N: T>(PhantomData<[T; N]>);
```

Comment:
It'd be nice if we could detect that `T` is a type param and instead of suggesting the annotation we suggested binding it with `PartialEq + Eq`.

Reply:
Ah, actually we ought to emit a specific error for this case: we don't currently accept using generic parameters as the types of const generics yet.